src/sentry/api/endpoints/organization_tags.py

```diff
@@ -27,6 +29,12 @@ def get(self, request: Request, organization) -> Response:
         except NoProjects:
             return Response([])
 
+        if request.GET.get("dataset"):
+            try:
+                Dataset(request.GET.get("dataset"))
+            except ValueError:
+                raise ParseError("Invalid dataset parameter")
```

Comment:
```suggestion
        try:
            dataset = Dataset(request.GET.get("dataset", "discover"))
        except ValueError:
            raise ParseError("Invalid dataset parameter")
```
If you pull out the dataset here you should be able to avoid having to keep the fallback "discover" aligned in all the gets everywhere else - line 46 should be able to be just `dataset=dataset,` and line 60 is a lot easier too

Reply:
Good call!